File: ProjectSR_Server/src/main/java/org/example/Classes/SocketMessage.java
package org.example.Classes;

import com.fasterxml.jackson.databind.ObjectMapper;

public class SocketMessage {
    private String senderId;
    private String senderUsername;
    private String type;
    private Object payload;
    private String receiverId;

    public SocketMessage() {}

    public SocketMessage(String senderId, String senderUsername, String type, Object payload, String receiverId) {
        this.senderId = senderId;
        this.senderUsername = senderUsername;
        this.type = type;
        this.payload = payload;
        this.receiverId = receiverId;
    }

    public String getSenderId() {
        return senderId;
    }

    public void setSenderId(String senderId) {
        this.senderId = senderId;
    }

    public String getSenderUsername() {
        return senderUsername;
    }

    public void setSenderUsername(String senderUsername) {
        this.senderUsername = senderUsername;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public Object getPayload() {
        return payload;
    }

    public void setPayload(Object payload) {
        this.payload = payload;
    }

    public String getReceiverId() {
        return receiverId;
    }

    public void setReceiverId(String receiverId) {
        this.receiverId = receiverId;
    }
}

File: ProjectSR_Client/ProjectSR_Client/Classes/SocketMessage.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ProjectSR_Client
{
    public class SocketMessage
    {
        public string SenderId { get; set; }
        public string SenderUsername { get; set; }
        public string Type { get; set; }
        public object Payload { get; set; }
        public string ReceiverId { get; set; }

        public SocketMessage(string senderId, string senderUsername, string type, object payload)
        {
            SenderId = senderId;
            SenderUsername = senderUsername;
            Type = type;
            Payload = payload;
        }

        public SocketMessage()
        {
        }


        public class Builder
        {
            private SocketMessage _message = new SocketMessage();

            public Builder SetSenderId(string senderId)
            {
                _message.SenderId = senderId;
                return this;
            }

            public Builder SetSenderUsername(string senderUsername)
            {
                _message.SenderUsername = senderUsername;
                return this;
            }

            public Builder SetType(string type)
            {
                _message.Type = type;
                return this;
            }

            public Builder SetPayload(object payload)
            {
                _message.Payload = payload;
                return this;
            }

            public Builder SetReceiverId(string receiverId)
            {
                _message.ReceiverId = receiverId;
                return this;
            }

            public SocketMessage Build()
            {
                return _message;
            }
        }
    }
}
